### PR TITLE
[lldb/test] Rework TestSwiftDeploymentTarget to not hardcode x86_64.

### PR DESCRIPTION
**Pull request by dcci**
This is preparation/generalization work for Apple Silicon.

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test

**Comment by fredriss** [comment]
why not put this on 5.3?

File: lldb/test/API/lang/swift/deployment_target/Makefile [hunk 1]
```diff
@@ -2,10 +2,12 @@ all: a.out dlopen_module
 
 include Makefile.rules
 
+ARCH := x86_64
+
 # This test only works on macOS 10.11+.
 
 a.out: main.swift libNewerTarget.dylib
-	$(SWIFTC) -target x86_64-apple-macosx10.10 -g -Onone $^ -lNewerTarget -L$(shell pwd) -o $@ $(SWIFTFLAGS) -I$(shell pwd) -Xfrontend -disable-target-os-checking
+	$(SWIFTC) -target $(ARCH)-apple-macosx10.10 -g -Onone $^ -lNewerTarget -L$(shell pwd) -o $@ $(SWIFTFLAGS) -I$(shell pwd) -Xfrontend -disable-target-os-checking
```
Comment (adrian-prantl) [inline lldb/test/API/lang/swift/deployment_target/Makefile:10]: This is good.

I think the Makefile would safer by doing a recursive make invocation that just sets `SWIFTFLAGS_EXTRAS=-Xfrontend -disable-target-os-checking and TRIPLE=$(ARCH)-apple-macosx10.10 LD_EXTRAS=-lNewerTarget`

```

**Comment by dcci** [comment]
> why not put this on 5.3?

Do we have an automerger?

File: lldb/test/API/lang/swift/deployment_target/Makefile (hunk 1 above)
Comment (dcci) [reply lldb/test/API/lang/swift/deployment_target/Makefile:10]: Can we do this as a follow up? It feels like a different improvement.

**Comment by fredriss** [comment]
> > why not put this on 5.3?
> 
> Do we have an automerger?

yes. You do need to manually put it on master-next though

**Comment by dcci** [comment]
I'll change the branch then.

**Comment by dcci** [comment]
I'll create a new PR to land this on 5.3

File: lldb/test/API/lang/swift/deployment_target/Makefile (hunk 1 above)
Comment (adrian-prantl) [reply lldb/test/API/lang/swift/deployment_target/Makefile:10]: Sure thing. I meant this to say: "the change is fine as is, if you want to go the extra mile, we can remove the meat of the build rules altogether by recursively calling make"

Comment (dcci) [reply lldb/test/API/lang/swift/deployment_target/Makefile:10]: Thanks.

**Pull request closed without merging**